src/System.Runtime.Extensions/tests/System/Runtime/CompilerServices/SwitchExpressionExceptionTests.cs

```diff
@@ -0,0 +1,47 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using Xunit;
+
+namespace System.Runtime.CompilerServices.Tests
+{
+    public class SwitchExpressionExceptionTests
+    {
+        [Fact]
+        public static void DefaultConstructor()
+        {
+            var ex = new SwitchExpressionException();
+
+            Assert.NotNull(ex.Message);
+        }
+
+        [Fact]
+        public static void Constructor_StringAsObjectArg()
+        {
+            string message = "stringInput";
+            var ex = new SwitchExpressionException(message);
+
+            Assert.NotEqual(message, ex.Message);
+            Assert.Equal(message, ex.UnmatchedValue);
+        }
+
+        [Fact]
+        public void UnmatchedValue()
+        {
+            var ex = new SwitchExpressionException(34);
+            Assert.Equal(34, ex.UnmatchedValue);
+
+            var data = new byte[] { 1, 2, 3 };
+            ex = new SwitchExpressionException(data);
+            Assert.Equal(data, ex.UnmatchedValue);
+
+            ex = new SwitchExpressionException(true);
+            Assert.Equal(true, ex.UnmatchedValue);
+
+            ex = new SwitchExpressionException("34");
+            Assert.Equal("34", ex.UnmatchedValue);
+        }
+    }
+}
```

Comment:
You should also update `src\System.Runtime.Serialization.Formatters\tests\BinaryFormatterTestData.cs` to make sure serialization is tested. @ViktorHofer can help if needed.

Reply:
@maryamariyan as you pinged me offline because of this, what help do you need here?

Reply:
where can I find instructions on how to add test in BinaryFormatterTestData for testing serialization for SwitchExpressionException?

Do you have a sample PR I can follow?

Reply:
replied offline.